core/src/main/java/org/infinispan/distribution/LocalizedCacheTopology.java

```diff
@@ -0,0 +1,190 @@
+package org.infinispan.distribution;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.infinispan.commons.hash.MurmurHash3;
+import org.infinispan.commons.util.Immutables;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.distribution.ch.ConsistentHash;
+import org.infinispan.distribution.ch.KeyPartitioner;
+import org.infinispan.distribution.ch.impl.ReplicatedConsistentHash;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.topology.CacheTopology;
+
+/**
+ * Extends {@link CacheTopology} with information about keys owned by the local node.
+ *
+ * @author Dan Berindei
+ * @since 9.0
+ */
+public class LocalizedCacheTopology extends CacheTopology {
```

Comment:
I think you're tying the LocalizedCacheTopology to a cacheMode too much; `CacheTopology` should be kept at the owners-level abstraction.
I'll try to comment where should particular methods belong.
Please think about scattered cache mode as the close-future newcomer; that should modify as few places as possible.